sound/soc/sof/sof-priv.h

```diff
@@ -357,6 +357,13 @@ struct snd_sof_dai {
 	struct list_head list;	/* list in sdev dai list */
 };
 
+/* DSP Core state */
+enum sof_dsp_state {
+	SOF_DSP_D3, /* DSP in reset state */
+	SOF_DSP_D0I3, /* DSP Running in D0i3(low power) state*/
+	SOF_DSP_D0I0, /* DSP Running in D0 state */
```

Comment:
why is this needed?
D3/D0I0 are already pm_runtime states. I don't know how you'd synchronize the value of this enum with the pm_runtime core.

Reply:
I added D3 and D0I0 to make the states integrated and easier for transition in state machine. Let me think about how to combine it with the system device PM states.